### ISSUE TITLE
Issue with pairwise conflict detection 

### ISSUE THREAD

**Issue by RubenJacobse** [TUDelft-CNS-ATM/bluesky#194]
I found an issue in the asas module where for a conflicting aircraft pair (a, b) the state-based conflict detection only detects the conflict from one 'side', i.e. it detects a conflict for (b, a) but not for (a, b). This issue seems to have a bigger impact at higher latitudes, and results in 'single-sided' resolution manoeuvres.

The root cause seems to be in the following code block of StateBasedCD.py (and possibly also the equivalent casas implementation):
```python
    # qdlst is for [i,j] qdr from i to j, from perception of ADSB and own coordinates
    qdr, dist = geo.qdrdist_matrix(np.mat(ownship.lat), np.mat(ownship.lon),
                                   np.mat(intruder.lat), np.mat(intruder.lon))

    # Convert back to array to allow element-wise array multiplications later on
    # Convert to meters and add large value to own/own pairs
    qdr = np.array(qdr)
    dist = np.array(dist) * nm + 1e9 * I

    # Calculate horizontal closest point of approach (CPA)
    qdrrad = np.radians(qdr)
    dx = dist * np.sin(qdrrad)  # is pos j rel to i
    dy = dist * np.cos(qdrrad)  # is pos j rel to i

 ...

    tcpa = -(du * dx + dv * dy) / dv2 + 1e9 * I

    # Calculate distance^2 at CPA (minimum distance^2)
    dcpa2 = np.abs(dist * dist - tcpa * tcpa * dv2)
```
Due to the curvature of the Earth the qdrs from a->b and b->a differ by an amount that is **not** equal to 180 degrees. When calculating the approximate cartesian distance components `dx` and `dy` between aircraft this means that for each 'side' of the pair a different sine and cosine are used. As a result the approximated values `tcpa` and `dcpa2` for the pair (a, b) can differ significantly from those of the pair (b, a).

In my BlueSky fork I have replaced the `qdr` value from `qdrdist_matrix()` with that of `kwikdistqdr_matrix` (see: RubenJacobse/bluesky@8aec61183fa7b3eff3fbeb501319216411884018). Since kwikdist uses a cartesian approximation the qdrs a->b and b->a differ by 180 degrees.
```python
    # qdlst is for [i,j] qdr from i to j, from perception of ADSB and own coordinates
    _, dist = geo.qdrdist_matrix(np.mat(ownship.lat), np.mat(ownship.lon),
                                 np.mat(intruder.lat), np.mat(intruder.lon))

    # qdrdist_matrix() uses WGS84 which means that qdr 1->2 is not the opposite of
    # qdr 2->1 which means their sine and cosine are different. This gives problems
    # in calculating dx and dy between pairs, since now dx,dy 1->2 are different
    # magnitudes from dx,dy 2->1 ...
    # Using kwikqdrdist_matrix() ensures that these relative distance approximations
    # are expected to remain within small margin of each other.
    qdr, _ = geo.kwikqdrdist_matrix(np.mat(ownship.lat), np.mat(ownship.lon),
                                    np.mat(intruder.lat), np.mat(intruder.lon))
```
It looks like this fix solves (or at least reduces) the issue of one-sided detection, but it's not a very elegant solution... I'm interested in hearing other peoples' opinions on this!

Example scenario files:
[confpair_issue_higher_latitude.scn.txt](https://github.com/TUDelft-CNS-ATM/bluesky/files/3173466/confpair_issue_higher_latitude.scn.txt)
[confpair_issue_near_equator.scn.txt](https://github.com/TUDelft-CNS-ATM/bluesky/files/3173467/confpair_issue_near_equator.scn.txt)

**Comment by jooste** [comment]
The solution actually might be to only use flat-earth estimates, as state-based CD is already based on a flat-earth assumption.

**Comment by RubenJacobse** [comment]
The interesting thing is that when I do this (use `qdr, dist = geo.kwikqdrdist_matrix()`) it seems that conflicts are never detected at all (regardless of latitude)...

**Comment by jooste** [comment]
This was due to a scaling bug. Fixed in 50362c9.

**Comment by RubenJacobse** [comment]
I would say that the combination of the above fix and only using the `kwikqdrdist_matrix()` function in StateBasedCD.py solves this issue. 

Created #198 to merge this into master.
